ui/app/utils/classes/rolling-array.js

```diff
@@ -0,0 +1,45 @@
+// An array with a max length.
+//
+// When max length is surpassed, items are removed from
+// the front of the array.
+
+// Using Classes to extend Array is unsupported in Babel so this less
+// ideal approach is taken: https://babeljs.io/docs/en/caveats#classes
```

Comment:
Are there any additional implications since ember has prototype extensions on? Seems like you would still be able to call ArrayProxy methods on the returned array, is that right? And is it a concern? (tried to think of a way to prevent that but nothing's coming to mind)

Reply:
There are, which I have addressed in a future branch. Mostly `addObject` needs to behave like push does.